src/pubtools/pulplib/_impl/client/client.py

```diff
@@ -1094,25 +1101,37 @@ def log_existing_repo(exception):
             raise exception
 
         def check_repo(repo_on_server):
+            # evolve some fields that don't have to be set before repository creation
+            # but they're typically set automatically after creation call which results in
+            # inequality between `repo_on_server` and `repo`
+            dists = [evolve(item, repo_id=repo_id) for item in repo.distributors]
+            repo_updated = evolve(
+                repo, relative_url=repo_on_server.relative_url, distributors=dists
+            )
             try:
                 assert (
-                    repo_on_server == repo
+                    repo_on_server == repo_updated
                 ), "Repo exists on server with unexpected values"
             except AssertionError:
                 if importer:
-                    for attr in ["type_id", "config"]:
-                        expected = getattr(repo.importer, attr)
-                        current = getattr(repo_on_server.importer, attr)
-                        if expected != current:
-                            LOG.error(
-                                "Repository %s contains wrong importer %s\n"
-                                "\t expected: %s\n"
-                                "\t current: %s\n",
-                                repo_id,
-                                attr,
-                                expected,
-                                current,
+                    self._check_unit(
+                        "importer",
+                        ["type_id", "config"],
+                        repo_updated.importer,
+                        repo_on_server.importer,
+                        repo_updated.id,
+                    )
+                for expected_item in repo_updated.distributors:
```

Comment:
This will log only the difference in the fields of a specific distributor type e.g. the difference in the fields for say cdn_distributor. 
However, there won't by any log if there are different set of distributors in the repos e.g. repo_on_server has yum and cdn distributors but the incoming or repo_updated has iso and cdn distributors. Only cdn distributors will be compared an logged but difference in presence of iso and yum distributors will be ignored while logging.

Reply:
added logging of missing and extra distributors, not super detailed but i guess good enough to see what's wrong.